crates/polars-time/src/chunkedarray/rolling_window/dispatch.rs

```diff
@@ -305,34 +329,52 @@ pub trait SeriesOpsTime: AsSeries {
         by: &Series,
         options: RollingOptionsDynamicWindow,
     ) -> PolarsResult<Series> {
-        let s = self.as_series().clone();
+        let original_type = self.as_series().dtype();
+        let mut s = self.as_series().clone();
+        if s.dtype().is_bool() {
+            s = s.cast(&DataType::UInt32)?;
+        }
+
         with_match_physical_numeric_polars_type!(s.dtype(), |$T| {
             let ca: &ChunkedArray<$T> = s.as_ref().as_ref().as_ref();
-            rolling_agg_by(
+            let out = rolling_agg_by(
                 ca,
                 by,
                 options,
                 &super::rolling_kernels::no_nulls::rolling_max,
-            )
+            )?;
+            if original_type.is_bool() {
+                out.cast(original_type)
+            } else {
+                Ok(out)
+            }
         })
     }
 
     /// Apply a rolling max to a Series.
     #[cfg(feature = "rolling_window")]
     fn rolling_max(&self, options: RollingOptionsFixedWindow) -> PolarsResult<Series> {
+        let original_type = self.as_series().dtype();
         let mut s = self.as_series().clone();
         if options.weights.is_some() {
             s = s.to_float()?;
+        } else if s.dtype().is_bool() {
```

Comment:
I really don't think we should cast here, but rather have a specialize algorithm for booleans.